routers/routes/goget.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package routes
+
+import (
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
+	"github.com/unknwon/com"
+)
+
+func goGet(ctx *context.Context) {
+	if ctx.Req.Method != "GET" || ctx.Query("go-get") != "1" || len(ctx.Req.URL.Query()) > 1 {
```

Comment:
Wouldn't `len(ctx.Req.URL.Query()) > 1` always be true for `?go-get=1`

Reply:
> 1 not > 0 ... this is so that it only fires if there is only 1 query param - go-get=1

Reply:
Aha, sorry I was thinking it returned the query-string. I see it returns `url.Values`.